File: Task-3/task3_ws/src/count_word/scripts/input_words.py
#!/usr/bin/python3

import rospy, sys
from count_word.srv import *


def count_words(usr_input):

    # node
    rospy.init_node("count_words_client", anonymous=True)

    # waiting service
    rospy.wait_for_service("count_words")
    
    # handling connection errors
    try:
        # service and request
        count_words_service = rospy.ServiceProxy("count_words", CountWord)
        request = CountWordRequest()
        request.words = usr_input
        response = count_words_service(request)
        return response.count

    except rospy.ServiceException as e:
        print("Service call failed %s" % e)


# handle user input
def usage():
    return "%s String " % sys.argv[0]


if __name__ == "__main__":
    while not rospy.is_shutdown():
        if len(sys.argv) == 1:
            print("Please input your sentence.")
            usr_input = input()
            print("Requesting to count number of words in:", usr_input)
            word_count = count_words(usr_input)
            print("Number of words:", word_count)
            print("\n")
        else:
            print("Usage: %s <String>" %sys.arg[0])
            sys.exit(1)


"""     if len(sys.argv) == 2:
       usr_input = sys.argv[1]
       print("Requesting to count number of words in:", usr_input)
       word_count = count_words(usr_input)
       print("Number of words:", word_count)
    else:
        print("Usage: %s <String>" %sys.arg[0])
        sys.exit(1) """

File: custom srv/task3_ws/src/count_words/src/input_words.cpp
#include <ros/ros.h>
#include <count_words/CountWord.h>
#include <iostream>
#include <string>

using namespace ros;

int main(int argc, char **argv)
{
    // node
    init(argc, argv, "count_words_client", init_options::AnonymousName);
    NodeHandle nh;

    while (ok())
    {

        // inputting
        std::string usr_input;
        if (argc == 1)
        {
            ROS_INFO("Please input your sentence.");
            std::getline(std::cin, usr_input);
            ROS_INFO("Requesting to count number of words in: %s", usr_input.c_str());
        }
        else
        {
            ROS_INFO("Usage: node doesn't accept arguments.");
            return 1;
        }
        spinOnce();

        // service and request
        ServiceClient client = nh.serviceClient<count_words::CountWord>("count_words");
        client.waitForExistence();

        // service
        count_words::CountWord srv;
        srv.request.words = usr_input;

        // handling calling errors
        if (client.call(srv))
            ROS_INFO("Number of words: %d\n", srv.response.count);
        else
        {
            ROS_INFO("Failed to call service count_words\n");
            return 1;
        }
    }

    return 0;
}